src/types/base.ts

```diff
@@ -0,0 +1,110 @@
+export enum MyInfoSource {
+  GovtVerified = '1',
+  UserProvided = '2',
+  NotApplicable = '3',
+  SingPassVerified = '4'
+}
+
+export enum MyInfoDataClassification {
+  Confidential = 'C'
+}
+
+type SourceProp<T> = {
+  source: T
+}
+
+// APPLICABILITY
+// Certain fields may not be applicable to specific groups of people. For example, CPF
+// or Residential Status is not applicable to foreigners, and Workpass is not applicable to Singapore
+// citizens and PRs.
+
+// For a full reference, see https://www.ndi-api.gov.sg/library/myinfo/implementation-myinfo-data
+export type MyInfoNotApplicable = SourceProp<MyInfoSource.NotApplicable>
+
+export type MyInfoApplicable = {
+  lastupdated: string
+  classification: MyInfoDataClassification.Confidential
+} & SourceProp<MyInfoSource.GovtVerified | MyInfoSource.UserProvided | MyInfoSource.SingPassVerified>
+
+type UnavailableProp<T> = {
+  unavailable: T
+}
+
+export type MyInfoUnavailableField = MyInfoApplicable & UnavailableProp<true>
+
+export type MyInfoAvailableMetadata = MyInfoApplicable & Partial<UnavailableProp<undefined>> // For convenience
+
+export type MyInfoField<T> = MyInfoNotApplicable | MyInfoUnavailableField |
```

Comment:
should `MyInfoNotApplicable` be here, since we only want it to apply to specific fields?

Reply:
oh yes, it was in my last commit but i forgot to push it